crystalfontz/dbus/map.py

```diff
@@ -0,0 +1,347 @@
+
+
+class ReadLcdMemoryM:
+    t: ClassVar[str] = t(AddressM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        address: int, timeout: float, retry_times: int
+    ) -> Tuple[int, Optional[float], Optional[int]]:
+        return (address, TimeoutM.load(timeout), RetryTimesM.load(retry_times))
+
+
+class LcdMemoryM:
+    t: ClassVar[str] = t("q", BytesM)
+
+    @staticmethod
+    def load(obj: Tuple[int, List[int]]) -> LcdMemory:
+        address, buff = obj
+        return LcdMemory(address, BytesM.load(buff))
+
+    @staticmethod
+    def dump(memory: LcdMemory) -> Tuple[int, List[int]]:
+        return (memory.address, BytesM.dump(memory.data))
+
+
+class SetCursorPositionM:
+    t: ClassVar[str] = t(PositionM, PositionM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        row: int, column: int, timeout: float, retry_times: int
+    ) -> Tuple[int, int, Optional[float], Optional[int]]:
+        return (row, column, TimeoutM.load(timeout), RetryTimesM.load(retry_times))
+
+
+CURSOR_STYLES: Dict[int, CursorStyle] = {style.value: style for style in CursorStyle}
+
+
+class CursorStyleM:
+    t: ClassVar[str] = "q"
+
+    @staticmethod
+    def load(style: int) -> CursorStyle:
+        return CURSOR_STYLES[style]
+
+    @staticmethod
+    def dump(style: CursorStyle) -> int:
+        return style.value
+
+
+class SetCursorStyleM:
+    t: ClassVar[str] = t(CursorStyleM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        style: int, timeout: float, retry_times: int
+    ) -> Tuple[CursorStyle, Optional[float], Optional[int]]:
+        return (
+            CursorStyleM.load(style),
+            TimeoutM.load(timeout),
+            RetryTimesM.load(retry_times),
+        )
+
+
+class SetContrastM:
+    t: ClassVar[str] = t("d", TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        contrast: float, timeout: float, retry_times: int
+    ) -> Tuple[float, Optional[float], Optional[int]]:
+        return (contrast, TimeoutM.load(timeout), RetryTimesM.load(retry_times))
+
+
+class SetBacklightM:
+    t: ClassVar[str] = t("dd", TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        lcd_brightness: float,
+        keypad_brightness: float,
+        timeout: float,
+        retry_times: int,
+    ) -> Tuple[float, Optional[float], Optional[float], Optional[int]]:
+        return (
+            lcd_brightness,
+            OptFloatM.load(keypad_brightness),
+            TimeoutM.load(timeout),
+            RetryTimesM.load(retry_times),
+        )
+
+
+class ReadDowDeviceInformationM:
+    t: ClassVar[str] = t(IndexM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def load(
+        index: int, timeout: float, retry_times: int
+    ) -> Tuple[int, Optional[float], Optional[int]]:
+        return (index, TimeoutM.load(timeout), RetryTimesM.load(retry_times))
+
+
+class DowDeviceInformationM:
+    t: ClassVar[str] = t(IndexM, BytesM)
```

Comment:
The API currently returns bytes here, but I think an int would be more appropriate. I filed #64 to capture this. I'd like to solve before I merge this.